src/blockstore/fs.ts

```diff
@@ -10,9 +9,9 @@ export class FsBlockStore extends BaseBlockstore implements Blockstore {
   _opened: boolean
   _opening?: Promise<void>
 
-  constructor () {
+  constructor (path: string) {
     super()
-    this.path = `${os.tmpdir()}/${(parseInt(String(Math.random() * 1e9), 10)).toString() + Date.now()}`
+    this.path = path
```

Comment:
This will be a breaking change release for users bringing their own Blockstore on pack. I think we should do this change, but with retro compatibility. Let's default to the original value here if no path is provided?

Reply:
I applied a change that should prevent custom blockstores from breaking, and still allow the clean up after a kill.